spec/formatting.md

```diff
@@ -14,9 +14,11 @@ To resolve the value of a Variable,
 its Name is used to identify either a local variable,
 or a variable defined elsewhere.
 If a local variable and an externally defined one use the same name,
-the local variable takes precedence.
+the local variable takes precedence after its variable declaration.
+A variable declaration may refer in its expression
+to a previous or external variable with the same name as the declaration's target.
```

Comment:
I'm not sure what "target" means here, and it isn't defined; can this be rephrased as something like:

"In a variable declaration, `let v = e`, `e` may refer to `v`. While resolving `e`, the meaning of `v` is the meaning of the most recently defined local or external variable named `v`."

It's hard to be more precise than that without talking about lexical scope and environments. In fact, for that reason, my personal preference is to disallow name shadowing. If name shadowing is allowed, and the spec stays at this level of formality, it will be harder to validate whether implementations comply with the spec.